### PR TITLE
CLOUD-45388 aws module, credential handling

### PR DESCRIPTION
**Pull request by bbihari**
@doktoric

### CODE REVIEW

File: cloud-api/src/main/java/com/sequenceiq/cloudbreak/cloud/notification/PersistenceNotifier.java [hunk 1]
```diff
@@ -1,16 +1,17 @@
 package com.sequenceiq.cloudbreak.cloud.notification;
 
 import com.sequenceiq.cloudbreak.cloud.event.context.CloudContext;
-import com.sequenceiq.cloudbreak.cloud.model.CloudResource;
+import com.sequenceiq.cloudbreak.cloud.model.DynamicModel;
 
 import reactor.rx.Promise;
 
-public interface PersistenceNotifier<T> {
+public interface PersistenceNotifier<E extends DynamicModel, T> {
```
Comment (akanto) [inline cloud-api/src/main/java/com/sequenceiq/cloudbreak/cloud/notification/PersistenceNotifier.java]: Why dynamic model extension is needed?

File: cloud-aws/build.gradle [hunk 2]
```diff
@@ -26,8 +26,11 @@ dependencies {
   compile group: 'commons-codec',                 name: 'commons-codec',                  version: '1.10'
   compile group: 'com.google.guava',              name: 'guava',                          version: guavaVersion
   compile (group: 'com.amazonaws',                name: 'aws-java-sdk',                   version: '1.10.11')      { exclude group: 'commons-logging' }
+  compile group: 'net.sf.ehcache',                name: 'ehcache',                        version: '2.8.3'
```
Comment (akanto) [inline cloud-aws/build.gradle]: ehcache dependency for aws modul?

File: cloud-aws/build.gradle [hunk 3]
```diff
@@ -26,8 +26,11 @@ dependencies {
...
 
   testCompile group: 'org.mockito',               name: 'mockito-all',                    version: '1.10.19'
   testCompile group: 'junit',                     name: 'junit',                          version: '4.12'
+  testCompile group: 'org.springframework.boot',  name: 'spring-boot-starter-test',       version: '1.1.8.RELEASE'
+  compile     group: 'org.hibernate',             name: 'hibernate-validator',            version: '5.1.1.Final'
```
Comment (akanto) [inline cloud-aws/build.gradle]: we shall not introduce any hibernate dependency even if it is a validator.

File: cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/AwsResourceConnector.java [hunk 4]
```diff
@@ -219,7 +214,8 @@ private String getEbsSnapshotIdIfNeeded(AuthenticatedContext ac, CloudStack clou
                     .withSize(SNAPSHOT_VOLUME_SIZE)
                     .withAvailabilityZone(availabilityZonesResult.getAvailabilityZones().get(0).getZoneName())
                     .withEncrypted(true));
-           /* EbsVolumeContext ebsVolumeContext = new EbsVolumeContext(stack, volumeResult.getVolume().getVolumeId());
+            /*
```
Comment (akanto) [inline cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/AwsResourceConnector.java]: something is wrong with the formatter

File: cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/AwsResourceConnector.java [hunk 5]
```diff
@@ -370,28 +368,28 @@ private void releaseReservedIp(CloudStack stack, AmazonEC2Client client) {
...
-       //         instanceGroupByInstanceGroupName.getNodeCount() + instanceCount);
-       // AutoScalingGroupReadyContext asGroupReady = new AutoScalingGroupReadyContext(stack, asGroupName, requiredInstances);
-       // LOGGER.info("Polling Auto Scaling group until new instances are ready. [stack: {}, asGroup: {}]", stack.getId(), asGroupName);
-       // PollingResult pollingResult = autoScalingGroupReadyPollingService
-       //         .pollWithTimeout(asGroupStatusCheckerTask, asGroupReady, POLLING_INTERVAL, MAX_POLLING_ATTEMPTS);
-       // if (!isSuccess(pollingResult)) {
-       //     throw new AwsResourceException("Failed to create CloudFormation stack, because polling reached an invalid end state.");
-       // }
+        // amazonASClient.updateAutoScalingGroup(new UpdateAutoScalingGroupRequest()
+        //         .withAutoScalingGroupName(asGroupName)
```
Comment (akanto) [inline cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/AwsResourceConnector.java]: extra whitespaces are introduced by the formatter

**Comment by doktoric** [comment]
I will fix these issues on the branch.

**Pull request merged**